src/docs/frontend/using-styled-components.mdx

```diff
@@ -0,0 +1,221 @@
+  color: ${p => p.theme.purple};
+  box-shadow: ${p => p.theme.dropShadowHeavy};
+`;
+```
+
+## Dynamic Properties
+
+<Alert level="info">
+  Most of the time dynamic properties are not needed!
+</Alert>
+
+You can pass in custom props to your custom component and use that to set styles. This does create overhead at runtime and should be avoided. What follows is an example of how to pass custom props, and how to avoid doing so.
+
+### Avoid Dynamic Props
+```tsx
+import styled from '@emotion/styled';
+
+// ❌ Try to avoid this
+const Label = styled('label')<{isDisabled: boolean; isError: boolean}>`
+  color: ${p => p.isDisabled
+    ? p.theme.gray200
+    : p.isError
+    ? p.theme.error400
+    : inherit
+  };
+`;
+
+return <Label disabled={false} isError={true}>There is an error</Label>;
+```
+
+Choose one of the strategies below instead.
+
+### Use Data Selectors
+
+Data selectors can be an easy way to set styles based on simple/pre-defined state. When defining these attributes don't forget that browser have built-in selectors like `:disabled`, `:focus`, `:first-child` and so on.
+
+```tsx
+import styled from '@emotion/styled';
+
+// ✅ Use a data attribute to pass in your state, and leverage CSS selectors:
+const Label = styled('label')`
+  color: inherit;
+  &[data-is-error="true"] {
+    color: ${p => p.theme.error400};
+  }
+  
+  &:disabled {
+    color: ${p => p.theme.gray200};
+  }
+`;
+
+return <Label disabled={false} data-is-error={true}>There is an error</Label>;
+```
+
+### Use CSS Variables
+
+For more flexible values, you can use CSS variables instead of pre-defined data-attributes for some value types.
+
+```tsx
+import styled from '@emotion/styled';
+
+// ✅ Define css variables for overridable properties:
+const Label = styled('label')`
+  color: var(--color, ${p => p.theme.textColor});
+`;
+
+const Error = styled('div')`
+  --color: ${p => p.theme.red400};
+`;
+const Info = styled('div')`
+  --color: ${p => p.theme.blue400};
+`;
+
+const label = <Label>There is an error</Label>;
+return isError
+  ? <Error>{label}</Error>
+  : <Info>{label}</Info>;
+```
+
+### Use Style Attributes
+
+The `style` attribute is possible to use, but the values are not linted. Avoid setting too many values at a time for readability.
+
+```tsx
+import styled from '@emotion/styled';
+
+// ✅ Don't be afraid of inline styles for one-off values
+const Grid = styled('div')`
+  display: grid;
+  grid-template-areas:
+    'logo search'
+    'sidebar main';
+  
+  @media (max-width: ${p => p.theme.breakpoints.small}) {
+    grid-template:
+        'logo' auto
+        'search' auto
+        'main' auto / 1fr;
+
+    /* Hide sidebar, which is in an auto-column */
+    grid-auto-columns: 0px;
+  }
+`;
+
+return (
+  <Grid>
+    <HomeLink style={{gridArea: 'logo'}} />
+    <SearchForm style={{gridArea: 'search'}} />
+    <Navigation style={{gridArea: 'sidebar'}} />
+    <DataTable style={{gridArea: 'main'}} />
+  </Grid>
+);
+```
+
+### Imperatively Set Styles
```

Comment:
Underrated! Scary to jump out of React's default rendering model, but this comes up a lot!